### ISSUE TITLE
syntax highliting issues with some syntax constructs like `(foo \a -> do -- comment)` or complex nested Types

### ISSUE THREAD

**Issue by rvion** [nwolverson/atom-ide-purescript#104]
🍏  with short string (< 108 char:

![image](https://cloud.githubusercontent.com/assets/2150990/16845258/933a15a8-49e8-11e6-8d00-fe074a1e39b9.png)

🔴  with long  (> 108 char) string:

![image](https://cloud.githubusercontent.com/assets/2150990/16845302/b11b8a34-49e8-11e6-9061-de1305b43eac.png)

🔴  bug appears both with and without wrap enabled:

![image](https://cloud.githubusercontent.com/assets/2150990/16845321/c3251d08-49e8-11e6-9df9-e7d75ca4a6e8.png)

**Comment by rvion** [comment]
I tried to isolate the bug:

🍏  without `--` after the `do`
![image](https://cloud.githubusercontent.com/assets/2150990/16845467/495e7770-49e9-11e6-84b8-26b6529ee0a1.png)

🔴 with `--` after a `do` blocl

![image](https://cloud.githubusercontent.com/assets/2150990/16845474/4f33b584-49e9-11e6-9007-23717cf543f4.png)

**Comment by nwolverson** [comment]
Interesting. There's an atom issue (design choice) that highlighting stops at a certain point in long lines https://github.com/atom/atom/issues/1667 - presume the fact that we get stuck inside a string is a consequence of the same thing.

Presume there's nothing we can do here, if indeed that's the thing. Suggested workaround is you can insert a comment to close the string without breaking your actual file:

```
--"
```

**Comment by nwolverson** [comment]
Closing this as this package doesn't do syntax highlighting. That is the domain of https://github.com/purescript-contrib/atom-language-purescript. I've raised an issue there that should fix this specific issue (https://github.com/purescript-contrib/atom-language-purescript/issues/18). Though it is surprising that the comment beforehand affects this in any way. The same highlighting is used for GitHub purescript fenced code blocks but in that case long lines break down in a different way that doesn't seem to get stuck in a string in this way.

It might be worth raising any other specific instances of this on that repo where we can improve matters a bit - though we won't avoid the fundamental atom limitation.
